sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -593,15 +605,34 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
   private def genCodeWithSet(ctx: CodegenContext, ev: ExprCode): ExprCode = {
     nullSafeCodeGen(ctx, ev, c => {
       val setTerm = ctx.addReferenceObj("set", set)
+      val hasNaNValue = ctx.addReferenceObj("hasNaN", hasNaN)
```

Comment:
`hasNaN` is a boolean, we can just use it to generate code
```
          |} else if (${isNaN(c)}) {
          |  ${ev.value} =  $hasNaN;
          |}
```

Reply:
Done